src/main/java/org/broadinstitute/consent/http/mail/SendGridAPI.java

```diff
@@ -48,15 +55,61 @@ public class SendGridAPI {
     private final DataCustodianApprovalMessage dataCustodianApprovalMessage = new DataCustodianApprovalMessage();
     private final DatasetApprovedMessage datasetApprovedMessage = new DatasetApprovedMessage();
     private final DatasetDeniedMessage datasetDeniedMessage = new DatasetDeniedMessage();
+    private final UserDAO userDAO;
 
-    public SendGridAPI(MailConfiguration config) {
-        this.fromAccount = config.getGoogleAccount();
-        this.sendGrid = new SendGrid(config.getSendGridApiKey());
-        this.activateEmailNotifications = config.isActivateEmailNotifications();
+    public SendGridAPI(MailConfiguration config, UserDAO userDAO) {
+        setFromAccount(config.getGoogleAccount());
+        setSendGrid(new SendGrid(config.getSendGridApiKey()));
+        setActivateEmailNotifications(config.isActivateEmailNotifications());
+        this.userDAO = userDAO;
     }
 
-    private Optional<Response> sendMessage(Mail message) {
-        if (activateEmailNotifications) {
+    private void setFromAccount(String fromAccount) {
+        this.fromAccount = fromAccount;
+    }
+
+    @VisibleForTesting
+    public void setSendGrid(SendGrid sendGrid) {
+        this.sendGrid = sendGrid;
+    }
+
+    private void setActivateEmailNotifications(Boolean activateEmailNotifications) {
+        this.activateEmailNotifications = activateEmailNotifications;
+    }
+
+    /**
+     * Determine if the user we are sending an email to has set their preference
+     * to false or not. Users who have been disabled like this should never receive
+     * an email.
+     *
+     * @param message The mail message
+     * @return False if the user has explicitly disabled email, True otherwise.
+     */
+    private boolean findUserEmailPreference(Mail message) {
+        List<Personalization> personalizations = message.getPersonalization();
+        Optional<Personalization> toEmail = personalizations.stream().findFirst();
+        if (toEmail.isPresent()) {
+            List<Email> tos = toEmail.get().getTos();
+            // When we construct a mail message, we always send it to just one individual
+            // so that we can track emails at that level of granularity.
+            if (!tos.isEmpty()) {
+                User user = userDAO.findUserByEmail(tos.get(0).getEmail());
+                if (Objects.isNull(user)) {
+                    logger.error("Unknown user: " + tos.get(0).getEmail());
+                    return false;
+                }
+                if (Objects.isNull(user.getEmailPreference())) {
+                    return true;
+                }
+                return user.getEmailPreference();
+            }
+        }
+        return false;
+    }
+
+    public Optional<Response> sendMessage(Mail message) {
+        boolean userEmailPreference = findUserEmailPreference(message);
+        if (activateEmailNotifications && userEmailPreference) {
```

Comment:
Would you want to log the case where the email would have been sent but the user's preference to suppress was set at the time?

Reply:
Good point, that's likely worth an `info` log, will do.